Message error for duplicate permissions on bulk edit

le0m opened the pull request:
This PR adds error details for bulk edit of permissions.

The new error message shows to the user which objects already had the permissions.

le0m committed:
- fix: add message error with details for duplicate permissions when bulk-editing

le0m committed:
- fix: bad english
- feat: added translation string for error message

batopa commented on bedita-app/app_controller.php:
```diff
@@ -1304,14 +1304,39 @@ public function addPermissionsObjects()
+            $duplicatePermissions = array();
             $this->Transaction->begin();
 
-            foreach ($objectIds as $id) {
+            foreach ($objectIds as $objectId) {
                 $permission = ClassRegistry::init('Permission');
-                $permission->add($id, $this->data['Permission']);
+                
+                foreach ($this->data['Permission'] as $permissionData) {
+                    if ($permission->isPermissionSet($objectId, $permissionData)) {
+                        if (!array_key_exists($objectId, $duplicatePermissions)) {
+                            $duplicatePermissions[$objectId] = array();
+                        }
+                        
+                        $duplicatePermissions[$objectId][] = $permissionData['name'];
+                    } else {
+                        $permission->add($objectId, array($permissionData));
+                    }
+                }
             }
 
             $this->Transaction->commit();
```
If `$permission->add()` fails all transaction is rollback.
Maybe we should remove the transaction here.

`Permission::isPermissionSet()` check just the `permissions.flag`, so you should check againts `flag` and `ugid` (`group_id` related to group name you have in the data array) and `'switch' => 'group'`.

Something as:

```php
$Permission = ClassRegistry::init('Permission');
$Group = ClassRegistry::init('Group');

foreach ($objectIds as $objectId) {
    foreach ($this->data['Permission'] as $permissionData) {
        $ugid = $group->field('id', array('name' => $permissionData['name']));
        $countPerms = $Permission->find('count', array(
            'conditions' => array(
                'switch' => 'group',
                'ugid' => $ugid,
                'flag' => $permissionData['flag'],
            ),
        ));
        if ($countPerms > 0) {
            $duplicatePermissions[$objectId][] = $permissionData['name']; // don't need check on key exists
            continue;
        }
        
        try {
            $Permission->add($objectId, array($permissionData));
        } catch (Exception $e) {
            // trace errros for view???
            $this->log('Error saving permission'); // more info is better
        }
    }
}
```

le0m replied:
I think we can leave the transaction, It's ok to rollback if It fails for any other reason. If an exception is raised, the transaction is rolled back and `AppController::handleError()` writes info to logs and shows details to the user.

Thanks for your snippet, I'll change the code. I guess I should add `object_id => $objectId` to the `$Permission->find(...)` conditions, right?

batopa replied:
correct, the condition on `object_id` is missing

le0m committed:
- fix: proper check for duplicate permissions
- refactor: remove transaction and collect all errors
- style: preserve newline and space chars in message template

le0m committed:
- feat: log to file
- chore: translated warning messages

manuelzane commented:
Metterei nel `bedita.css` per la classe `messageDetail`
questi 2 valori:
```
left: 300px;
min-height: 140px;
```
la classe penso sia usata anche per altri messaggi ma non mi sembra un problema modificarla per tutti.

le0m committed:
- style: message detail CSS
